src/modules/schnorr_adaptor/tests_impl.h

```diff
@@ -0,0 +1,957 @@
+
+    CHECK(secp256k1_schnorr_adaptor_adapt(CTX, sig64, sig, secadaptor) == 1);
+    CHECK_ILLEGAL(CTX, secp256k1_schnorr_adaptor_adapt(CTX, NULL, sig, secadaptor));
+    CHECK_ILLEGAL(CTX, secp256k1_schnorr_adaptor_adapt(CTX, sig64, NULL, secadaptor));
+    CHECK_ILLEGAL(CTX, secp256k1_schnorr_adaptor_adapt(CTX, sig64, sig, NULL));
+
+    CHECK(secp256k1_schnorr_adaptor_adapt(CTX, sig64, sig, secadaptor) == 1);
+    CHECK(secp256k1_schnorr_adaptor_extract_sec(CTX, adaptor, sig, sig64) == 1);
+    CHECK_ILLEGAL(CTX, secp256k1_schnorr_adaptor_extract_sec(CTX, NULL, sig, sig64));
+    CHECK_ILLEGAL(CTX, secp256k1_schnorr_adaptor_extract_sec(CTX, adaptor, NULL, sig64));
+    CHECK_ILLEGAL(CTX, secp256k1_schnorr_adaptor_extract_sec(CTX, adaptor, sig, NULL));
+
+}
+
+/* Helper function for schnorr_adaptor_vectors
+ * Signs the message and checks that it's the same as expected_sig. */
+void test_schnorr_adaptor_vectors_check_presigning(const unsigned char *sk, const unsigned char *pk_serialized, const unsigned char *aux_rand, const unsigned char *msg32, const unsigned char *adaptor, const unsigned char *expected_sig) {
+    unsigned char sig[65];
+    unsigned char t[33];
+    secp256k1_keypair keypair;
+    secp256k1_xonly_pubkey pk, pk_expected;
+
+    CHECK(secp256k1_keypair_create(CTX, &keypair, sk));
+    CHECK(secp256k1_schnorr_adaptor_presign(CTX, sig, msg32, &keypair, adaptor, aux_rand));
+    CHECK(secp256k1_memcmp_var(sig, expected_sig, 65) == 0);
+
+    CHECK(secp256k1_xonly_pubkey_parse(CTX, &pk_expected, pk_serialized));
+    CHECK(secp256k1_keypair_xonly_pub(CTX, &pk, NULL, &keypair));
+    CHECK(secp256k1_memcmp_var(&pk, &pk_expected, sizeof(pk)) == 0);
+    CHECK(secp256k1_schnorr_adaptor_extract(CTX, t, sig, msg32, &pk));
+    CHECK(secp256k1_memcmp_var(t, adaptor, 33) == 0);
+}
+
+/* Helper function for schnorr_adaptor_vectors
+ * Extracts the adaptor point and checks if it returns the same value as expected. */
+ void test_schnorr_adaptor_vectors_check_extract(const unsigned char *pk_serialized, const unsigned char *msg32, const unsigned char *sig, const unsigned char *expected_t, int expected) {
+    secp256k1_xonly_pubkey pk;
+    unsigned char t[33];
+
+    CHECK(secp256k1_xonly_pubkey_parse(CTX, &pk, pk_serialized));
+    CHECK(secp256k1_schnorr_adaptor_extract(CTX, t, sig, msg32, &pk));
+    CHECK(expected == (secp256k1_memcmp_var(t, expected_t, 33) == 0));
+ }
+
+/* Helper function for schnorr_adaptor_vectors
+ * Adapt a Schnorr adptor signature to a valid Schnorr signature
+ * and checks if it is the same as expected_sig64. */
+void test_schnorr_adaptor_vectors_check_adapt(const unsigned char *sig, const unsigned char *secadaptor, const unsigned char *expected_sig64, int expected) {
+    unsigned char sig64[64];
+
+    CHECK(secp256k1_schnorr_adaptor_adapt(CTX, sig64, sig, secadaptor));
+    CHECK(expected == (secp256k1_memcmp_var(sig64, expected_sig64, 64) == 0));
+}
```

Comment:
I don't think we need to check this is a valid schnorr sig here since we think that adapt function would return a valid schnorr sig and the schnorr sig in the test vector should be valid. But in the python specification when creating the schnorr sig, we just use the adapt to create the schnorr sig in the test vectors. So I suggest we check if valid schnorr sig in the python specification when creating the test vectors

Reply:
The current testing method does this:

- extracts `expected_bip340_sig` and `test_result` from the test vector CSV file
- Then it checks the following:
    - **Case 1:** If `test_result == TRUE`
        - `adapt(presig, secadaptor)` output should match `expected_bip340_sig`
    - **Case 2:** If `test_result == FALSE`
        - `adapt(presig, secadaptor)` output shouldn’t match `expected_bip340_sig`

This way of testing is logically correct. However, it relies on the assumption that the `expected_bip340_sig` mentioned in the test vector is correct (i.e., passes `schnorrsig_verify`).

I think it would be nice to avoid this correctness assumption by directly calling `schnorrsig_verify` on the `adapt(presig, secadaptor)` output.

If we do this, then the new testing method would do:

- extract `test_result` from the test vector CSV file
- Then it will checks:
    - **Case 1:** If `test_result == TRUE`
        - `adapt(presig, secadaptor)` output should pass `schnorr_verify`
    - **Case 2:** If `test_result == FALSE`
        - `adapt(presig, secadaptor)` output should fail `schnorr_verify`

Reply:
Now two months after I initially wrote this, I don't know anymore why I suggested this. I agree with @siv2r that this rules out some sort of human error but that's a very minor thing. In particular, we need a message and public key to verify against which may not be available and complicate things.

Maybe I commented on the wrong line. I definitely wanted to suggest to call `schnorrsig_extract` in the randomized "correctness" tests (and not the test vector tests) where we create a schnorrsig via adapt (such as `test_schnorr_adaptor_adapt_extract_sec`). So imo we can resolve this convo.

Reply:
Will add schnorrsig calls in the tests